socorro/tests/stage_submitter/test_submitter.py

```diff
@@ -0,0 +1,585 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import gzip
+import json
+import random
+
+from markus.testing import MetricsMock
+import pytest
+
+from socorro import settings
+from socorro.stage_submitter.submitter import (
+    get_payload_compressed,
+    get_payload_type,
+    remove_collector_keys,
+    SubmitterApp,
+)
+
+
+def get_app():
+    app = SubmitterApp()
+    app.set_up()
+    return app
+
+
+def generate_s3_key(kind, crash_id):
+    """Generates the key in S3 for this object kind
```

Comment:
this isn't just used for s3, i suggest we rename it:
```suggestion
def generate_storage_key(kind, crash_id):
    """Generates the key in cloud storage for this object kind
```

Reply:
Whoops--this is a copy/paste fail.